src/lightning/pytorch/loops/dataloader/prediction_loop.py

```diff
@@ -145,18 +150,16 @@ def _on_predict_epoch_end(self) -> Optional[_PREDICT_OUTPUT]:
         Returns:
             the results for all dataloaders
         """
-        results = self.predictions
-
-        self.trainer._call_callback_hooks("on_predict_epoch_end", results)
-        self.trainer._call_lightning_module_hook("on_predict_epoch_end", results)
+        self.trainer._call_callback_hooks("on_predict_epoch_end")
+        self.trainer._call_lightning_module_hook("on_predict_epoch_end")
 
         if self.return_predictions:
-            return results[0] if self.num_dataloaders == 1 else results
+            return self.predictions
```

Comment:
This goes through the property I added, which does the same logic.

The property was added so that the users can have the same convenience of automatically getting the first index if there's just one dataloader